File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    
    n1 = 0
    n2 = 0
    
    for word in goal:
        if n1 < len(cards1) and word == cards1[n1]:
            n1 += 1
            
        elif n2 < len(cards2) and word == cards2[n2]:
            n2 += 1
            
        else:
            return "No"
        
    return "Yes"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int n1 = 0;
        int n2 = 0;
        
        for(int i = 0; i < goal.length ; i++){
            if(n1 < cards1.length && goal[i].equals(cards1[n1])){
                n1++;
            }else if(n2 < cards2.length && goal[i].equals(cards2[n2])){
                n2++;
            }else {
                return "No";
            }
        }
        
        return "Yes";
    }
}